src/workers/python/PythonWorker.worker.ts

```diff
@@ -65,17 +65,44 @@ class PythonWorker extends Backend<PyodideExtras> {
         await (this.pyodide as any).loadPackage("micropip");
     }
 
+    /**
+     * Private helper to parse Pyodide loading messages into Loading events
+     * @param {string} m The Pyodide loading message
+     */
+    private importMessageCallback(m: string): void {
+        if (m.startsWith("Loading ")) {
+            const packageStart = "Loading ".length;
+            // message optionally contains a "from <source>" part
+            const packageEnd = m.includes(" from") ? m.indexOf(" from") : m.length;
+            this.onEvent({
+                type: BackendEventType.Loading,
+                contentType: "application/json",
+                data: {
+                    loading: true,
+                    modules: m.slice(packageStart, packageEnd).split(", ")
+                }
+            });
+        } else if (m.startsWith("Loaded")) {
+            this.onEvent({
+                type: BackendEventType.Loading,
+                data: {
+                    loading: false,
+                    modules: m.slice("Loaded ".length).split(", ")
+                },
+                contentType: "application/json"
+            });
+        }
+    }
+
     /**
      * Helper method to install imports and prevent race conditions with double downloading
      * @param {string} code The code containing import statements
-     * @param {boolean} ignoreMissing Whether to ignore failures on missing modules
      */
-    private async installImports(code: string, ignoreMissing: boolean): Promise<void> {
+    private async installImports(code: string): Promise<void> {
         if (this.installPromise == null) {
-            this.installPromise = this.papyros.install_imports.callKwargs({
-                source_code: code,
-                ignore_missing: ignoreMissing
-            });
+            this.installPromise = (this.pyodide as any)
+                .loadPackagesFromImports(code,
```

Comment:
Why this change?

Reply:
This is specifically to have access to the message callback option provided by Pyodide. Doesn't Pyodide also perform micropip-installs if needed, or is that specific to pyodide-worker-runner?

Reply:
I don't think it does, see https://github.com/pyodide/pyodide/issues/1470#issuecomment-907815342

You can check by seeing what happens when you import something that isn't pyodide-specific. Right now it sort of looks like `loadPackagesFromImports` will miss it but the new system might still catch it at runtime.

Reply:
Should I look into opening a PR on pyodide-worker-runner to accept the callbacks instead? Then we can have the best of both worlds.